apps/io-profile-async/src/functions/__tests__/expired-sessions-discoverer.test.ts

```diff
@@ -0,0 +1,550 @@
+      expect(
+        mockExpiredUserSessionsQueueRepository.sendExpiredUserSession
+      ).toHaveBeenCalledWith(item.queuePayload, item.itemTimeoutInSeconds);
+      expect(E.isRight(result)).toBe(true);
+      expect(trackEventMock).not.toHaveBeenCalled();
+    });
+
+    it("should fail when the flag update fails", async () => {
+      const aCosmosError = { kind: "COSMOS_ERROR_RESPONSE" } as CosmosErrors;
+
+      updateExpiredSessionNotificationFlagMock.mockImplementationOnce(() =>
+        TE.left(aCosmosError)
+      );
+      const result = await processItem(item)(baseDeps)();
+
+      expect(
+        mockSessionNotificationsRepository.updateExpiredSessionNotificationFlag
+      ).toHaveBeenCalledWith(
+        item.retrievedDbItem.id,
+        item.retrievedDbItem.expiredAt,
+        true
+      );
+      expect(
+        mockExpiredUserSessionsQueueRepository.sendExpiredUserSession
+      ).not.toHaveBeenCalled();
+      expect(result).toStrictEqual(
+        E.left(
+          new TransientError(
+            `Error updating expired session flag(EXPIRED_SESSION): ${aCosmosError.kind}`
+          )
+        )
+      );
+    });
+
+    it("should fail when send to queue fails and flag revert fails", async () => {
+      const anError = new Error("Send to queue failed");
+      const aCosmosError = { kind: "COSMOS_ERROR_RESPONSE" } as CosmosErrors;
+
+      updateExpiredSessionNotificationFlagMock
+        .mockImplementationOnce(() => TE.of(void 0))
+        .mockImplementationOnce(() => TE.left(aCosmosError));
+
+      sendExpiredUserSessionMock.mockImplementationOnce(() => TE.left(anError));
+
+      const result = await processItem(item)(baseDeps)();
+      expect(
+        mockSessionNotificationsRepository.updateExpiredSessionNotificationFlag
+      ).toHaveBeenNthCalledWith(
+        1,
+        item.retrievedDbItem.id,
+        item.retrievedDbItem.expiredAt,
+        true
+      );
+      expect(
+        mockExpiredUserSessionsQueueRepository.sendExpiredUserSession
+      ).toHaveBeenCalledWith(item.queuePayload, item.itemTimeoutInSeconds);
+      expect(
+        mockSessionNotificationsRepository.updateExpiredSessionNotificationFlag
+      ).toHaveBeenNthCalledWith(
+        2,
+        item.retrievedDbItem.id,
+        item.retrievedDbItem.expiredAt,
+        false
+      );
+      expect(trackEventMock).toHaveBeenCalledWith({
+        name:
+          "io.citizen-auth.prof-async.expired-sessions-discoverer.permanent.revert-failure",
+        properties: {
+          // eslint-disable-next-line no-underscore-dangle
+          itemDbSelf: item.retrievedDbItem._self,
+          message:
+            "Error reverting expired session flag(EXPIRED_SESSION) after Queue write failure"
+        },
+        tagOverrides: {
+          samplingEnabled: "false"
+        }
+      });
+      expect(E.isRight(result)).toBe(true);
+    });
+  });
+
+  describe("processChunk", () => {
```

Comment:
For future reference: to avoid duplicating already tested scenarios, we should mock `processItem` in this suite of tests.